rfcs/0002-source-acl/README.md

```diff
@@ -0,0 +1,186 @@
+# RFC-0002 Access control for source references
+
+## Summary
+
+Cross-namespace references to Flux sources should be subject to
+Access Control Lists (ACLs) as defined by the owner of a particular source.
+
+Similar to [Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/),
+Flux ACLs define policies for restricting the access to the source artifact server based on the
+caller's namespace.
+
+## Motivation
+
+This proposal tries to solve the "cross-namespace references side-step namespace isolation" issue (explained in
+[RFC-0001](https://github.com/fluxcd/flux2/tree/main/rfcs/0001-authorization#cross-namespace-references-side-step-namespace-isolation)).
+
+As of [version 0.25](https://github.com/fluxcd/flux2/releases/tag/v0.25.0) (Ian 2022),
+Flux allows for `Kustomizations` and `HelmReleases` to reference sources in different namespaces.
+This poses a serious security risk for multi-tenant environments as Flux does not prevent tenants from accessing
+known sources outside of their namespaces.
+
+Flux does not allow for an `ImageUpdateAutomation` to reference a `GitRepository` in a different namespace.
+This means users have to copy the Git auth secret and the `GitRepository` object in all namespaces
+where `ImageUpdateAutomations` are used. This poses a serious security risk for multi-tenant environments,
+as tenants could use the Git secret to push changes to repositories by impersonating Flux.
+
+Flux allows for `ImagePolicies` to reference `ImageRepositories` in a different namespace only 
+if the ACL present on the `ImageRepository` grants access to the namespace where the `ImagePolicy` is.
+This has been implemented in
+[fluxcd/image-reflector-controller#162](https://github.com/fluxcd/image-reflector-controller/pull/162).
+
+Flux should be consistent when dealing with cross-namespace references by extending the
+Image Policy/Repository approach to all the other APIs.
+
+### Goals
+
+- Allow source owners to choose which sources are shared and with which namespaces.
+- Allow cluster admins to enforce source ACLs.
+
+### Non-Goals
+
+- Enforce source ACLs by default. 
```

Comment:
This will maintain backwards compatibility as far as I understand. Let's be explicit about that here, too: This RFC doesn't introduce backwards-incompatible changes.